pdata/pcommon/value.go

```diff
@@ -111,6 +114,12 @@ func NewValueSlice() Value {
 	return newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_ArrayValue{ArrayValue: &otlpcommon.ArrayValue{}}}, &state)
 }
 
+// NewValueTemplate creates a new Value with the given string value.
+func NewValueTemplate(v string) Value {
+	state := internal.StateMutable
+	return newValue(&otlpcommon.AnyValue{Value: &otlpcommon.AnyValue_StringValue{StringValue: v}}, &state)
```

Comment:
@dmitryax maybe question here, should we rather define a full new datatype for the template type, or reuse the string one, since template is not a real-world datatype. Thanks!